api/v1alpha1/k8ssandracluster_types.go

```diff
@@ -129,6 +140,53 @@ type EmbeddedObjectMeta struct {
 	Annotations map[string]string `json:"annotations,omitempty"`
 }
 
+// TODO Implement Stringer interface. It will helpful for debugging and testing.
+type CassandraConfig struct {
+	CassandraYaml *CassandraYaml `json:"cassandraYaml,omitempty"`
+
+	JvmOptions *JvmOptions `json:"jvmOptions,omitempty"`
+}
+
+type CassandraYaml struct {
```

Comment:
I wonder how future-proof this will be? There might be differences already between C* 3.11 and C* 4.0, and who knows what surprises C* 4.1 and higher will bring wrt to config options... How do you think we would handle these discrepancies?

Reply:
I think this is more generalised issue with the config-builder problem we have. And it's not just the CassandraYaml section, but also JvmOptions. JvmOptions depend on the used JDK version and the selected GC.  Those properties do not make much sense when used with ZGC or Shenandoah (both of which we'll want to promote with 4.0)

Reply:
>I wonder how future-proof this will be?

Is it not sufficient to something like this:

```
if version is 3.11.x 
    // apply 3.11.x specific settings
else 
    // apply 4.x specific settings
```

Can't we do something similar for JVM options? If we know the image that is being used then we should know which JVM is being used and should be able to apply settings based on that knowledge.

Right now the goal is just to provide parity with what we offer in the Helm charts.